compiler/sem/semexprs.nim

```diff
@@ -45,13 +45,14 @@ proc semOperand(c: PContext, n: PNode, flags: TExprFlags = {}): PNode =
   if result.typ != nil:
     # XXX tyGenericInst here?
     if result.typ.kind == tyProc and hasUnresolvedParams(result, {efOperand}):
-      result = c.config.newError(n, reportAst(rsemProcHasNoConcreteType, n))
+      result = c.config.newError(n, PAstDiag(kind: adSemProcHasNoConcreteType))
     elif result.typ.kind in {tyVar, tyLent}:
       result = newDeref(result)
   elif {efWantStmt, efAllowStmt} * flags != {}:
     result.typ = newTypeS(tyVoid, c)
   else:
-    result = c.config.newError(n, reportAst(rsemExpressionHasNoType, result))
+    # TODO: were passing `result` to the report and `n` to the error
```

Comment:
This comment seems to no longer apply?

Reply:
I was treating them as reminders, I used `TODO`, probably should have had it as `xxx`, the ambiguity is useful. Anyhow, they're not super informative, going to drop them.

Thanks.